config/vlan.py

```diff
@@ -182,103 +230,162 @@ def config_proxy_arp(db, vid, mode):
-
-        if (clicommon.interface_is_untagged_member(db.cfgdb, port) and untagged): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} is already untagged member!".format(port))
+    if ADHOC_VALIDATION:
+        for vid in vid_list:
+
+            # default vlan checker
+            if vid == 1:
+                ctx.fail("{} is default VLAN".format(vlan))
+
+            log.log_info("'vlan member add {} {}' executing...".format(vid, port))
+
+            if not clicommon.is_vlanid_in_range(vid):
+                ctx.fail("Invalid VLAN ID {} (2-4094)".format(vid))
+
+            vlan = 'Vlan{}'.format(vid)
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
+                log.log_info("{} does not exist".format(vlan))
+                ctx.fail("{} does not exist".format(vlan))
+
+            if clicommon.get_interface_naming_mode() == "alias":  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                alias = port
+                iface_alias_converter = clicommon.InterfaceAliasConverter(db)
+                port = iface_alias_converter.alias_to_name(alias)
+                if port is None:
+                    ctx.fail("cannot find port name for alias {}".format(alias))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if clicommon.is_port_mirror_dst_port(db.cfgdb, port):
+                ctx.fail("{} is configured as mirror destination port".format(port))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if clicommon.is_port_vlan_member(db.cfgdb, port, vlan):
+                log.log_info("{} is already a member of {}, Aborting!!!".format(port, vlan))
+                ctx.fail("{} is already a member of {}, Aborting!!!".format(port, vlan))
+                
+
+            if clicommon.is_valid_port(db.cfgdb, port):
+                is_port = True
+            elif clicommon.is_valid_portchannel(db.cfgdb, port):
+                is_port = False
+            else:
+                ctx.fail("{} does not exist".format(port))
+
+            portchannel_member_table = db.cfgdb.get_table('PORTCHANNEL_MEMBER')
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if (is_port and clicommon.interface_is_in_portchannel(portchannel_member_table, port)):
+                ctx.fail("{} is part of portchannel!".format(port))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if (clicommon.interface_is_untagged_member(db.cfgdb, port) and untagged):
+                ctx.fail("{} is already untagged member!".format(port))
+            
+            # checking mode status of port if its access, trunk or routed
+            if is_port:
+                port_data = config_db.get_entry('PORT',port)
+            
+            # if not port then is a port channel
+            elif not is_port:
+                port_data = config_db.get_entry('PORTCHANNEL',port)
+
+            if "mode" not in port_data: 
```

Comment:
>port_data [](http://example.com/codeflow?start=29&length=9)

PORT/mode is not yang modelled. So this PR will become breaking change, and will block this submodule move. I will raise a PR to revert this PR.

Reply:
@qiluo-msft , 
@venkatmahalingam @gechiang We have fixed YANG Model for this PR. The [YANG PR](https://github.com/sonic-net/sonic-buildimage/pull/13580) is failing as it needs supported utilties sourcecode. One posssible solution is that we merge https://github.com/sonic-net/sonic-utilities/pull/2419 utilities code and then we merge this PR. All dependencies will be be solved. 
Other solution is that we remove MUST condition from this PR for instance in sonic-vlan.yang for Vlan membership for only on access and trunk port and merge this PR and merge utilities source code. After that i will add a new PR with must condition check on sonic-vlan membership.It is requested to please help in this matter. Thanks!